packages/webview_flutter/webview_flutter_wkwebview/example/ios/RunnerTests/FWFWebViewHostApiTests.m

```diff
@@ -351,4 +351,40 @@ - (void)testEvaluateJavaScript {
   XCTAssertEqualObjects(returnValue, @"result");
   XCTAssertNil(returnError);
 }
+
+- (void)testEvaluateJavaScriptReturnsNSErrorData {
+  FWFWebView *mockWebView = OCMClassMock([FWFWebView class]);
+
+  OCMStub([mockWebView
+      evaluateJavaScript:@"runJavaScript"
+       completionHandler:([OCMArg invokeBlockWithArgs:[NSNull null],
+                                                      [NSError errorWithDomain:@"errorDomain"
+                                                                          code:0
+                                                                      userInfo:@{
+                                                                        NSLocalizedDescriptionKey :
+                                                                            @"description"
+                                                                      }],
+                                                      nil])]);
+
+  FWFInstanceManager *instanceManager = [[FWFInstanceManager alloc] init];
+  [instanceManager addDartCreatedInstance:mockWebView withIdentifier:0];
+
+  FWFWebViewHostApiImpl *hostAPI =
+      [[FWFWebViewHostApiImpl alloc] initWithInstanceManager:instanceManager];
+
+  NSString __block *returnValue;
+  FlutterError __block *returnError;
+  [hostAPI evaluateJavaScriptForWebViewWithIdentifier:@0
+                                     javaScriptString:@"runJavaScript"
+                                           completion:^(id result, FlutterError *error) {
+                                             returnValue = result;
+                                             returnError = error;
+                                           }];
+
+  XCTAssertNil(returnValue);
+  FWFNSErrorData *errorData = returnError.details;
```

Comment:
Do you want to actually assert the type as well? It looks like all the asserts here would actually pass even if the object were actually an `NSError`, since it has all of the same methods.